lib/docker.js

```diff
@@ -135,24 +140,50 @@ function createDockerFile() {
 
     // set the deploy target
     baseImg = targets[pkg.deploy.target];
+    if (!baseImg || baseImg === '') {
+        console.error('ERROR: You must specify a valid target!');
+        console.error('ERROR: Check the deploy stanza in package.json and targets.yaml');
+        process.exit(2);
+    }
+
     // get any additional packages that need to be installed
     Object.keys(pkg.deploy.dependencies).forEach(function(sys) {
-        if (sys !== '_all' && (sys === baseImg || (new RegExp(sys)).test(baseImg))) {
-            Array.prototype.push.apply(extraPkgs, pkg.deploy.dependencies[sys]);
+        if (sys === '_all' || (sys === baseImg || (new RegExp(sys)).test(baseImg))) {
+            pkg.deploy.dependencies[sys].forEach(function(pkg) {
+                if (typeof pkg === 'string') {
+                    extraPkgs.push(pkg);
+                } else if (typeof pkg === 'object') {
+                    if (!pkg.repo_url || !pkg.pool || !pkg.packages || !pkg.release) {
+                        console.error('ERROR: Incorrect dependency spec: ' + JSON.stringify(pkg));
+                    }
+                    customSourcePkgs.push(pkg);
+                } else {
+                    console.error('ERROR: Incorrect dependency spec: ' + pkg);
+                    process.exit(2);
+                }
+            });
         }
     });
-    Array.prototype.push.apply(extraPkgs, pkg.deploy.dependencies._all);
 
-    if (!baseImg || baseImg === '') {
-        console.error('ERROR: You must specify a valid target!');
-        console.error('ERROR: Check the deploy stanza in package.json and targets.yaml');
-        process.exit(2);
+    if (customSourcePkgs.some(function(customSourcePkgSpec) {
+            return /^https/.test(customSourcePkgSpec.repo_url);
+        })) {
+        extraPkgs.push('apt-transport-https');
     }
 
     contents = 'FROM ' + baseImg + '\n' +
-        'RUN apt-get update --fix-missing && apt-get install -y ' + extraPkgs.join(' ') +
+        'RUN apt-get update && apt-get install -y ' + extraPkgs.join(' ') +
         ' && rm -rf /var/lib/apt/lists/*\n';
 
+    customSourcePkgs.forEach(function(customSourcePkgSpec) {
+        contents += 'RUN echo "deb ' + customSourcePkgSpec.repo_url +
```

Comment:
@d00rman Unfortunately we can't do that, or I don't know how to achieve this. In `apt.wikimedia.org` the node version is 4.xxx and it depends on some npm version and gyp version which are not in jessie repo and not in apt.wikimedia repo, so if we try to install our normal deps after adding apt.wikimedia to the sources list - it fails. Would be awesome if you had an idea how to solve this.